samples/f77/demo_ftnlib.cpp

```diff
@@ -81,38 +93,34 @@ extern "C" {
             string fin = string(file, lenfile);
             string fth = string(id, lenid);
             trmodel = string(transport, lentr);
-            delete _gas;
-            _gas = new IdealGasMix(fin, fth);
+            _sol = newSolution(fin, fth, trmodel);
+            _gas = std::dynamic_pointer_cast<IdealGasPhase>(_sol->thermo());
+            _kin = std::dynamic_pointer_cast<GasKinetics>(_sol->kinetics());
+            _trans = _sol->transport();
         } catch (CanteraError& err) {
             handleError(err);
         }
-        try {
-            delete _trans;
-            _trans = newTransportMgr(trmodel,_gas,0);
-        } catch (CanteraError& err) {
-            _trans = newTransportMgr("",_gas,0);
-        }
     }
 
     /// integer function nElements()
-    integer nelements_()
+    int nelements_()
```

Comment:
Unlike `doublereal`, there is perhaps an argument for using a `typedef` for the integer type when working with Fortran, which is that someone could be using Fortran code that uses 8 byte integers and want to be able to define `integer` appropriately.

Reply:
reverting.